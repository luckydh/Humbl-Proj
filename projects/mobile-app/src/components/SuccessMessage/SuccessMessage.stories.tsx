import React from "react";
import { Meta, Story } from "@storybook/react/types-6-0";

import withPadding from "../../utils/withPadding";
import SuccessMessage, { SuccessMessageProps } from "./SuccessMessage";

export default {
  title: "Components/SuccessMessage",
  component: SuccessMessage,
  argTypes: {
    onClick: { action: "onClick" },
  },
  decorators: [withPadding],
} as Meta;

const Template: Story<SuccessMessageProps> = (args) => <SuccessMessage {...args} />;

export const Primary = Template.bind({});
Primary.args = {};
