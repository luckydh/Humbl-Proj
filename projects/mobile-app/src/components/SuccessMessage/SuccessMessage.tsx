import React, { FC } from "react";
import { Link } from "react-router-dom";
import { IonPage } from "@ionic/react";
import CheckMark from "components/CheckMark";
import { HomeIcon } from "assets/svgs/HomeIcon";
import { HumblLogoResizeable } from "assets/svgs/HumblLogoResizeable";

export interface SuccessMessageProps {
  message: string;
}

const SuccessMessage: FC<SuccessMessageProps> = ({ children, message }) => (
    <IonPage className="bg-profiles safe-area-top">
      <div className="flex flex-col h-full justify-between pt-12 pb-4">
        <div className="mx-auto max-w-xs px-4">
          <HumblLogoResizeable />
        </div>
        <div className="flex items-center flex-col relative w-full">
          <CheckMark noBackground />
          <div className="flex-1 flex flex-col flex-grow justify-center text-center text-white">
            <h1 className="font-medium text-center text-3xl mt-8 text-white px-12 tracking-tight leading-tight">
              {message}
            </h1>
            {children}
          </div>
        </div>
        <Link
          to="/home"
          replace={true}
          data-testid="return_home"
          className="border-2 border-white border-solid rounded-full self-center my-6">
          <HomeIcon className="w-12 h-12 text-white fill-current" />
        </Link>
      </div>
    </IonPage>
  );

export default SuccessMessage;
