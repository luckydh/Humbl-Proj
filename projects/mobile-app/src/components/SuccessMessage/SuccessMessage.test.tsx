import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { createMemoryHistory } from "history";
import { StaticRouter, Router } from "react-router-dom";

import SuccessMessage from "./SuccessMessage";

describe("SuccessMessage", () => {
  it("renders the given message if a merchant is passed through route state", () => {
    const locationState = {
      state: {
        merchant: {},
      },
    };

    const { queryByText, container } = render(
      <StaticRouter location={locationState}>
        <SuccessMessage message="Success message" />
      </StaticRouter>
    );

    expect(queryByText("Success message")).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it("returns to /home when the Home button is pressed", () => {
    const history = createMemoryHistory();

    history.push("/", {
      merchant: {},
    });

    const { getByTestId } = render(
      <Router history={history}>
        <SuccessMessage message="Success message" />
      </Router>
    );

    userEvent.click(getByTestId("return_home"));

    expect(history.location.pathname).toBe("/home");
  });
});
