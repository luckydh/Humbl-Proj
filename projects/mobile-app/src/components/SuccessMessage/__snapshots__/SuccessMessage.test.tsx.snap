// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessMessage renders the given message if a merchant is passed through route state 1`] = `
<div>
  <div
    class="ion-page bg-profiles safe-area-top"
  >
    <div
      class="flex flex-col h-full justify-between pt-12 pb-4"
    >
      <div
        class="mx-auto max-w-xs px-4"
      >
        <svg
          class="block w-full"
          viewBox="0 0 143 31"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#FFF"
            fill-rule="nonzero"
          >
            <path
              d="M18.418 1v11.484H7.151V1H0v29.794h7.151V18.335h11.267v12.459h7.259V1zM37.594 1v17.118a7.15 7.15 0 001.3 4.55 4.659 4.659 0 003.9 1.517 5.092 5.092 0 003.9-1.517 6.934 6.934 0 001.3-4.55V1h7.151v17.118c.09 2.45-.47 4.88-1.625 7.042a11.376 11.376 0 01-4.442 4.334 13.543 13.543 0 01-6.392 1.406c-2.18.06-4.339-.423-6.284-1.408a10.4 10.4 0 01-4.334-4.334 14.518 14.518 0 01-1.517-7.148V1h7.043zM93.39 1.217v29.577h-7.151V12.376l-6.392 18.527h-6.175L67.28 12.376v18.527h-7.151V1.217h8.776l8.017 20.585 7.907-20.585zM120.475 18.118a6.934 6.934 0 011.625 4.55 7.476 7.476 0 01-2.492 5.959 11.159 11.159 0 01-7.259 2.167H98.482V.998h13.543c2.5-.159 4.98.528 7.042 1.95a6.934 6.934 0 012.492 5.634 6.609 6.609 0 01-1.408 4.442 6.934 6.934 0 01-3.792 2.384 8.884 8.884 0 014.117 2.709m-14.952-4.875h4.767c2.492 0 3.792-1.083 3.792-3.25s-1.3-3.25-3.792-3.25h-4.658l-.109 6.5zm9.209 8.451a2.817 2.817 0 00-1.083-2.492 4.225 4.225 0 00-3.034-.867h-5.092v6.609h5.092c2.709 0 4.117-1.083 4.117-3.25M133.044 25.268h9.642v5.525h-16.793V1h7.151zM140.519 1a2.167 2.167 0 110 4.334 2.167 2.167 0 010-4.334m0 4.117a1.95 1.95 0 10-1.95-1.95 1.95 1.95 0 001.95 1.95m-.758-3.25h.975c.542 0 .867.217.867.758a.65.65 0 01-.65.65l.758 1.192h-.217l-.758-1.192h-.65v1.192h-.325v-2.6zm.217 1.192h.65c.433 0 .65-.108.65-.433s-.217-.542-.65-.542h-.65v.975z"
            />
          </g>
        </svg>
      </div>
      <div
        class="flex items-center flex-col relative w-full"
      >
        <div
          class="z-50 border-white p-12 rounded-full opacity-95"
          data-testid="checkmark"
          style="top: 0px; width: 219px; height: 219px; border-width: 7px;"
        >
          <svg
            class="w-full h-full"
            fill="none"
            height="87"
            viewBox="0 0 103 87"
            width="103"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M29.668 86.514h-.053a.608.608 0 01-.452-.194l-.038-.042L.773 58.28a.52.52 0 01-.136-.386l.009-.155-.012-.154a.62.62 0 01.176-.482l9.197-9.099.03-.033a.682.682 0 01.972-.032l.096.096 18.63 18.17L91.948.702a.688.688 0 01.978-.032l.082.084 9.21 9.113a.58.58 0 01.149.44l-.012.155.009.155a.525.525 0 01-.154.4L30.255 86.292a.707.707 0 01-.512.224l-.075-.002z"
              fill="#fff"
            />
          </svg>
        </div>
        <div
          class="flex-1 flex flex-col flex-grow justify-center text-center text-white"
        >
          <h1
            class="font-medium text-center text-3xl mt-8 text-white px-12 tracking-tight leading-tight"
          >
            Success message
          </h1>
        </div>
      </div>
      <a
        class="border-2 border-white border-solid rounded-full self-center my-6"
        data-testid="return_home"
        href="/home"
      >
        <svg
          class="w-12 h-12 text-white fill-current"
          height="31.268"
          viewBox="0 0 31.268 31.268"
          width="31.268"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.634 0a15.634 15.634 0 1015.634 15.634A15.634 15.634 0 0015.634 0zm6.3 21.213a.708.708 0 01-.7.7h-4.181v-4.188h-2.8v4.184h-4.182a.708.708 0 01-.7-.7V15.98a.171.171 0 01.005-.032.124.124 0 00.005-.032l6.268-5.168 6.272 5.168a.151.151 0 01.011.065zm1.24-6.989l-.678.807a.376.376 0 01-.226.12h-.033a.334.334 0 01-.226-.075l-6.363-5.282-6.391 5.345a.405.405 0 01-.262.075.371.371 0 01-.226-.12l-.677-.806a.356.356 0 01-.075-.257.315.315 0 01.12-.234l6.69-5.588a1.355 1.355 0 011.656 0l6.648 5.525a.316.316 0 01.12.234.353.353 0 01-.075.256z"
            fill="#fff"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
